docs/System.md

```diff
@@ -0,0 +1,93 @@
+# System Overview
+
+This document provides an overview of the Exo compilation process, as illustrated in Figure 1 of the PLDI'22 paper.
+
+![System overview](images/system-overview.png)
+
+The Exo compiler consists of a frontend and a backend, with user schedules applied in between. The input to the compiler is a set of Exo source files (`*.py`), and the output is generated C code (`*.c`).
+
+In this repository, folders are structured as follows:
+
+1. `src/exo` is where the core Exo implementation resides.
+    - **APIs.**
+      - `API.py` defines a stable API for top-level decorators (`proc`, `instr`, and `config`).
+      - `API_scheduling.py` defines a API for scheduling primitives.
+      - `API_cursors.py` defines a API for Cursors.
+    - **Standard libraries.** These could be user-defined, but we provide them for convenience.
+      - `libs/` contains some common memory definitions (`memories.py`) and custom malloc implementations. 
+      - `platforms/` contains instruction definitions that are part of the release.
+      - `stdlib/` contains user-level scheduling functions such as `vectorize`.
+    - Other files are implementation of Exo (e.g., `typecheck.py` implements typecheck), are not exposed to users.
+2. `apps/` contains some sample applications written in Exo.
+3. `dependencies/` contains submodules that Exo's apps and testing depends on.
+4. `examples/` contains examples of scheduling with Exo.
+5. `tests/` contains the Exo test suite.
+6. `docs/` contains additional Exo documentation.
+
+---
+
+## Core
+
+`src/exo/core` defines IRs used in Exo and other core implementations.
+- `LoopIR.py` is the main file that defines IRs (LoopIR, UAST, PAST), and their visitor functions (LoopIR_Do, LoopIR_Rewrite). 
+- `LoopIR_pprint.py` implements a printing procedure for the IRs defined in `LoopIR.py`.
+- `prelude.py` defines `Sym` and `Srcinfo`.
+
+User-defined features like config, externs, and Memory's parent class implementations are in `configs.py`, `extern.py`, and `memory.py`, respectively.
+
+`internal_cursors` defines cursor movements that are used internally by `LoopIR_scheduling` implementations of scheduling primitives.
```

Comment:
I feel like a bit more background is needed here for internal_cursors to make sense.